modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/hazelcast/HazelcastAgent.java

```diff
@@ -947,7 +947,9 @@ public void deleteMessageIdsByQueueName(String queueName) throws AndesException
             HashMap<String, TreeSet<Slot>> queueToSlotMap = wrapper.getStringListHashMap();
             if (queueToSlotMap != null) {
                 TreeSet<Slot> slotListForQueueOnNode = queueToSlotMap.get(queueName);
-                resultSet.addAll(slotListForQueueOnNode);
+                if(slotListForQueueOnNode != null) {
```

Comment:
Normally we flip this to avoid developer mistakes. Make it "null !=  slotListForQueueOnNode"